src/view/nova/lib/theme.ts

```diff
@@ -0,0 +1,394 @@
+import type {
+  ViewStyle,
+  TextStyle,
+  ImageStyle,
+  DimensionValue,
+  ColorValue,
+} from 'react-native'
+import {StyleSheet} from 'react-native'
+
+type StyleObject = ViewStyle & TextStyle & ImageStyle
+type StyleObjectProperties = keyof StyleObject
+type ColorProperties =
+  | 'color'
+  | 'backgroundColor'
+  | 'borderColor'
+  | 'borderTopColor'
+  | 'borderRightColor'
+  | 'borderBottomColor'
+  | 'borderLeftColor'
+type DimensionProperties =
+  | 'padding'
+  | 'paddingTop'
+  | 'paddingBottom'
+  | 'paddingLeft'
+  | 'paddingRight'
+  | 'paddingVertical'
+  | 'paddingHorizontal'
+  | 'margin'
+  | 'marginTop'
+  | 'marginBottom'
+  | 'marginLeft'
+  | 'marginRight'
+  | 'top'
+  | 'bottom'
+  | 'left'
+  | 'right'
+  | 'gap'
+  | 'rowGap'
+  | 'columnGap'
+
+export type Tokens = {
+  space?: DimensionValue[] | Record<string, DimensionValue>
+} & {
+  [Property in StyleObjectProperties]?: Record<
+    string,
+    Required<StyleObject>[Property]
+  >
+}
+export type Properties = Record<string, Readonly<StyleObjectProperties[]>>
+export type Macros<T extends Tokens> = Record<
+  string,
+  (value: any, tokens: T) => StyleObject
+>
+export type Breakpoints = {
+  [Breakpoint: string]: number
+}
+
+export type Styles<
+  T extends Tokens,
+  P extends Properties,
+  M extends Macros<T>,
+> = {
+  [Property in ColorProperties]?:
+    | keyof T['color']
+    | Omit<ColorValue, keyof T['color']>
+} & {
+  [Property in DimensionProperties]?: keyof T['space'] | DimensionValue
+} & {
+  [Property in Exclude<
+    StyleObjectProperties,
+    ColorProperties | DimensionProperties
+  >]?: keyof T[Property] | StyleObject[Property]
+} & {
+  // this empty P is to avoid circular references back to Properties type
+  [Property in keyof P]?: Styles<T, {}, M>[P[Property][0]]
+} & {
+  [Property in keyof M]?: M[Property] extends (
+    value: infer Value,
+    tokens: T,
+  ) => StyleObject
+    ? Value
+    : never
+}
+
+export type ResponsiveStyles<
+  T extends Tokens,
+  P extends Properties,
+  M extends Macros<T>,
+  B extends Breakpoints,
+> =
+  | Styles<T, P, M>
+  | {
+      [Breakpoint in keyof B]?: Styles<T, P, M>
+    }
+
+export type Theme<
+  T extends Tokens,
+  P extends Properties,
+  M extends Macros<T>,
+  B extends Breakpoints,
+> = {
+  config: {
+    tokens: T
+    properties: P
+    macros: M
+    breakpoints: B
+  }
+  style: <Props>(
+    props: ResponsiveStyles<T, P, M, B> & Omit<Props, keyof P>,
+    breakpoints?: (keyof B)[],
+  ) => {
+    styles: StyleObject
+    props: Props
+  }
+  getActiveBreakpoints: ({width}: {width: number}) => {
+    active: (keyof B)[]
+    current: keyof B
```

Comment:
Let's rename `current` to something else. React's lint rules assume `.current` means ref access which will trip them up in some cases. Also a common assumption for readers (I assumed this is a ref)